test/cancelable_operation_test.dart

```diff
@@ -387,6 +387,27 @@ void main() {
         expect(runThen().value, throwsA('error'));
         originalCompleter.operation.cancel();
       });
+
+      test('after completing with a future does not invoke `onValue`', () {
+        onValue = expectAsync1((_) => '', count: 0);
+        onCancel = null;
+        var operation = runThen();
+        var workCompleter = Completer<int>();
+        originalCompleter.complete(workCompleter.future);
+        originalCompleter.operation.cancel();
+        workCompleter.complete(0);
+        expect(operation.isCanceled, true);
+      });
```

Comment:
Is it worth adding a delay at the end, perhaps `await workCompleter.future;` to give the work-completer time to actually complete asynchronously, before declaring success.

Reply:
The final `expect` ensures that we have gotten past the point where `onValue` would have been called. I can add this to make the test more clearly correct.